src/main/java/com/salesforce/comdagen/model/Site.kt

```diff
@@ -10,26 +10,45 @@ package com.salesforce.comdagen.model
 import com.salesforce.comdagen.Comdagen
 import com.salesforce.comdagen.Configuration
 import com.salesforce.comdagen.SeedInheritance
+import com.salesforce.comdagen.XMLOutputProducer
 import com.salesforce.comdagen.config.*
 import com.salesforce.comdagen.generator.*
 import java.io.File
 
 class Site(
     private val internID: Int,
+    /**
+     * This is either a seed or a random number generated from the top level seed (indexed site generation)!
+     */
     private val seed: Long,
     private val defaults: SiteConfiguration?,
     private val config: SiteConfiguration,
     private val configDir: File
 ) {
+    init {
+        /*
+         * Add some private stats such as seed to the statistics companion object.
+         * A different way would be to change the visibility of the attributes or
```

Comment:
I'd go with the public seed over a static field hack any day. Let the `XMLOutputProducer` or an even more specialized class take care of this and keep that knowledge out of the `Site` class.